rhizome/host/lib/vm_setup.rb

```diff
@@ -83,8 +84,36 @@ def setup_networking(skip_persisted, gua, ip4, local_ip4, nics, ndp_needed, mult
     forwarding
   end
 
+  def manage_ip4_block_status(ip4, action)
+    ip_net = NetAddr::IPv4Net.parse(ip4).network.to_s
+    File.open(Util::NFTABLES_LOCK_FILE_PATH, File::RDWR | File::CREAT) do |lock|
+      lock.flock(File::LOCK_EX)
+      File.open(Util::NFTABLES_PATH, File::RDWR) do |nftfile|
+        nftfile.flock(File::LOCK_EX)
+
+        nft = nftfile.read
+        lines = nft.split("\n").map(&:strip)
+
+        commands = case action
+        when :unblock
+          lines.reject { _1.include?(ip_net) }
+        when :block
+          lines.insert(lines.index("type filter hook prerouting priority 0; policy accept;") + 1, "ip daddr #{ip_net} drop")
```

Comment:
What happens if the same line added here twice? Would it cause any problems? Do we need to deduplicate the lines here?

Reply:
Not really, also, it cannot happen if we didn't use the same ip twice.

Reply:
If this code executed twice due to a problem in future steps, it can happen, no? But if having same entry twice does cause any problem, then it is OK.